Controller/Payment/Webhook.php

```diff
@@ -0,0 +1,446 @@
+                        ]
+                    );
+                    header('Status: 400 Signature Verification failed', true, 400);
+                    exit;
+                }
+                switch ($post['event']) {
+                    case 'payment.authorized':
+                    case 'order.paid':
+                        return $this->orderPaid($post);
+                    default:
+                        return;
+                }
+            }
+        }
+    }
+
+    /**
+     * Order Paid webhook
+     *
+     * @param array $post
+     */
+    protected function orderPaid(array $post)
+    {
+        if ($this->config->getPaymentAction() === \Razorpay\Magento\Model\PaymentMethod::ACTION_AUTHORIZE
+          && $post['event'] === 'payment.authorized') {
+            $this->authorize($post);
+        }
+        if ($this->config->getPaymentAction() === \Razorpay\Magento\Model\PaymentMethod::ACTION_AUTHORIZE_CAPTURE
+          && $post['event'] === 'order.paid') {
+            $this->authorizeCapture($post);
+        }
+    }
+
+    /**
+     * Payment Authorized
+     *
+     * @param array $post
+     */
+    protected function authorize(array $post)
+    {
+        $this->logger->info("Razorpay Webhook Event(" . $post['event'] . ") processing Started.");
+        $paymentId      = $post['payload']['payment']['entity']['id'];
+        $rzpOrderId     = $post['payload']['payment']['entity']['order_id'];
+        $amountPaid     = $post['payload']['payment']['entity']['amount'];
+        $email          = $post['payload']['payment']['entity']['email'];
+        $contact        = $post['payload']['payment']['entity']['contact'];
+        try {
+            $rzpOrder       = $this->getRzpOrder($rzpOrderId);
+            $quoteId        = $rzpOrder->receipt;
+            $rzpOrderAmount = $rzpOrder->amount;
+            if (isset($quoteId) === false) {
+                $this->logger->info("Razorpay Webhook: Quote ID not set for Razorpay payment_id(:$paymentId)");
+                return;
+            }
+        } catch (\Razorpay\Api\Errors\Error $e) {
+            $this->logger->critical("Razorpay Webhook: fetching RZP order data(id:$rzpOrderId) "
+                                . "PaymentId:(:$paymentId) failed with error: ". $e->getMessage());
+            return;
+        } catch (\Exception $e) {
+            $this->logger->critical("Razorpay Webhook: fetching RZP order data(id:$rzpOrderId) "
+                                . "PaymentId:(:$paymentId) failed with error: ". $e->getMessage());
+            return;
+        }
+        try {
+            # fetch the related sales order and verify the payment ID with rzp payment id
+            # To avoid duplicate order entry for same quote
+            $collection = $this->objectManagement->get('Magento\Sales\Model\Order')
+            ->getCollection()
+            ->addFieldToSelect('entity_id')
+            ->addFilter('increment_id', $quoteId)
```

Comment:
@enishant , Variable name must be orderID. ( we are not using quote anymore here now).

Reply:
@ramth05 Updated variable name to orderId